packages/app/src/features/liquidity-flow/components/asset-select/asset-select.tsx

```diff
@@ -49,28 +48,29 @@ export function AssetSelect({ tokenNames, onChange, value }: Props) {
   return (
     <div>
       {renderButton()}
-      <Menu anchorEl={anchorEl} open={open} onClose={close}>
+      <S.Menu anchorEl={anchorEl} open={open} onClose={close}>
         {tokenNames.map((tokenName, index) => {
           const token = getTokenByTokenName(tokenName)
-          if (!token) return null
-          const { title, symbol, decimals } = token
+          const tokenBalance = depositBalances[index]?.balance.toNumber() || 0
+          if (!token || (!tokenBalance && isDeposit)) return null
```

Comment:
Maybe it's better to take the business logic out of the component and leave only the display.
And remove isDeposit prop

Reply:
refactored